### DISCUSSION TITLE
Kami/Yi App SD Card Recordings

### DISCUSSION THREAD

**Discussion by rpkilgore** [roleoroleo/yi-hack-Allwinner-v2#377] [General]
Has anyone else noticed that after you apply the hack, you can no longer view recordings on the sd card in the Kami or Yi Apps?
If you try to scroll back in the timeline to a recording or try to view on sd card it just jumps to live..
Thought maybe it was a permissions issue, but looks like the permissions are identical as far as I can tell.
Checked the mp4's using ffmepg and they look similar.  There is a very slight difference in bitrate, but don't think that's it.

Does the hack change anything with how the motion events are recorded?
Has anyone else noticed this?  If so have you found a fix for this?

Thanks,
rpkilgore

**Comment by ghost** [comment]
See https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/246#issuecomment-1041002228

**Reply by rpkilgore** [reply]
Are you saying that it was a timezone issue?  I'm in eastern time, the hack and the app are both set to EST, and the recordings match up to EST as well.

**Reply by roleoroleo** [reply]
I think there is an issue but I'm pretty sure that it depends on the model of the cam.

**Reply by rpkilgore** [reply]
So far I've seen it on the y28ga (2 of them), r30gb (2 of them), and r35gb (2 of them)
I'm testing and playing with a y28ga right now to see if i'm missing a tz setting somewhere.

**Reply by rpkilgore** [reply]
So, this is interesting... the date/time is correct everywhere except in ssh.  Where it's set to GMT instead of EST..  witch I could see as being a problem since the date/time stamps won't match what the app is expecting..  Even though it IS creating the correct time stamp using the folder structure and file name.. the actual date/time stamps of the files are in GMT. strange...

![snapshot](https://user-images.githubusercontent.com/19828182/156038794-ff82a5c7-d33f-4b58-ab9c-0fb6f8a1e098.png)
![systempage](https://user-images.githubusercontent.com/19828182/156038849-d5b82778-7701-42d0-bafa-5a3b1ddf8e62.png)
![ssh](https://user-images.githubusercontent.com/19828182/156038873-28f70a70-0edf-4624-9c73-13ca2fcbd8a8.png)
![Screenshot_20220228-125633](https://user-images.githubusercontent.com/19828182/156038895-8b90d203-1587-4a51-a3bb-d86acf4334e9.jpg)
![Screenshot_20220228-125659](https://user-images.githubusercontent.com/19828182/156038942-06430b2c-6391-4b92-b884-7438c0bd88d5.jpg)
Another interesting thing is that i can view those clips, but when you goto view on sd card it just jumps to live.  Also notice the recordings seem to be in the correct place on the timeline...  
How do I set the date/time on the cam in ssh so that it matches est?

**Reply by roleoroleo** [reply]
I will open a new thread about tz.

**Comment by rpkilgore** [comment]
So, this is strange, popped a fresh sd card in, and it does use the GMT time for the folder and filename so they match up, but in the app they show in eastern time.. so it does appear to be time stamp related.  Looks like the folder structure should match the file name date/time stamp...

![dirlist](https://user-images.githubusercontent.com/19828182/156050144-bb5adaf7-3c47-4f9a-a651-1e67044133f8.png)
![Screenshot_20220228-145123](https://user-images.githubusercontent.com/19828182/156050305-a3b64953-b396-44c1-a521-5586d8f535a0.jpg)

**Comment by ghost** [comment]
@rpkilgore  the 'TZ' export in the hack might also needs changing so that it only affects those programs that need it such as the web server. IIRC (which is doubtful) the TZ export was set in the system.sh file so that it was applied after dispatch and before rmm. I don't think Yi use TZ, at least not on the camera's I've used. With SSH I don't know if the TZ can be applied other than after it has launched, maybe one of the Linux people can shed some light. 

I hope that helps as I will be gone for some time.

**Comment by roleoroleo** [comment]
Please check this issue:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/378
I uploaded a new beta where I changed the way system.sh exports TZ.
Do you want to try it?

**Reply by rpkilgore** [reply]
Sure I'll jump over to that thread and test it out..
